app/custom_auth.py

```diff
@@ -0,0 +1,106 @@
+            
+            if not User.objects.filter(username=username).exists() and settings.CREATE_USER:
+                create_user(username)
+            
+            user = auth.authenticate(remote_user=username)
+            process_user_groups(user,request.META)
+            
+            if user:
+                request.user = user
+                auth.login(request, user)
+
+        except KeyError:
+            logger.info(f"LOG: User name header {settings.HEADER_AUTH_PROXY_USER} not found. Continue without user authenication." )
+         
+        response = self.get_response(request)
+        return response
+
+def process_user_groups(user,headers):
+        groups_from_header = get_user_groups_from_header(headers)
+        admin_group_name = settings.ADMIN_GROUP_NAME
+        admin_group = get_admin_group(admin_group_name)
+        
+        # Case 1: IS_SUPER flag takes precedence over HTTP_X_AUTH_PROXY_GROUPS header.
+        if settings.IS_SUPER:
+                if not user.groups.filter(name=admin_group_name).exists():
+                    user.groups.add(admin_group)
+                user.is_superuser = settings.IS_SUPER
+                user.save()
+                return 
+        
+        # Case 2: HTTP_X_AUTH_PROXY_GROUPS header contains Admin group. 
+        if  groups_from_header and admin_group_name in groups_from_header:
+                print ("HTTP_X_AUTH_PROXY_GROUPS header contains Admin group ")
+                if not user.groups.filter(name=admin_group_name).exists():
+                    user.groups.add(admin_group)
+                user.is_superuser = True
+                user.save()
+
+        # Case 3: HTTP_X_AUTH_PROXY_GROUPS header is absent or does not contain Admin group.
+        if not groups_from_header or not admin_group_name in groups_from_header:
+                print ("HTTP_X_AUTH_PROXY_GROUPS header does not contains Admin group ")
+                if user.groups.filter(name=admin_group_name).exists():
+                    user.groups.remove(admin_group)
+                user.is_superuser = False
+                user.save() 
+        
+def get_admin_group(admin_group):
+        if Group.objects.filter(name=admin_group).exists():
+            return Group.objects.get(name=admin_group)
+        else:
+            group = Group(name=admin_group)
+            group.save()
+            return group
+
+def get_user_groups_from_header(headers):
+    if settings.HEADER_AUTH_PROXY_GROUPS in headers:
+                return headers[settings.HEADER_AUTH_PROXY_GROUPS].split(',')
+
+def create_user(username):
+
+    try:
+        email =  f"{username}@noreply.com" 
+        user = User.objects.create_user(username=username,email=email)
+        user.is_superuser=settings.IS_SUPER
+        user.save()
+        return user
+
+    except Exception as ex:
+        logger.info(f"LOG: User creation failed. Continue without creating user. Details: {ex}" )
```

Comment:
Should use `logger.exception` so that full exception information including stacktrace/etc is automatically included in the log message.